SRMSG00071: Invalid channel configuration

### Describe the bug

Hello,
I have a QuarkusTest in my project which uses Kafka and the KafkaCompanionResource. At the moment I use the quarkus version 3.28.5 and get the following error if I start the test:

```
Caused by: java.lang.IllegalArgumentException: SRMSG00071: Invalid channel configuration -  the `connector` attribute must be set for channel `treats`
	at io.smallrye.reactive.messaging.providers.impl.ConnectorConfig.lambda$getConnectorAttribute$0(ConnectorConfig.java:65)
	at java.base/java.util.Optional.orElseThrow(Optional.java:403)
	at io.smallrye.reactive.messaging.providers.impl.ConnectorConfig.lambda$getConnectorAttribute$1(ConnectorConfig.java:65)
	at java.base/java.util.Optional.orElseGet(Optional.java:364)
	at io.smallrye.reactive.messaging.providers.impl.ConnectorConfig.getConnectorAttribute(ConnectorConfig.java:64)
	at io.smallrye.reactive.messaging.providers.impl.ConnectorConfig.<init>(ConnectorConfig.java:59)
	at io.smallrye.reactive.messaging.providers.impl.ConfiguredChannelFactory.lambda$extractConfigurationFor$0(ConfiguredChannelFactory.java:95)
	at java.base/java.lang.Iterable.forEach(Iterable.java:75)
```

I use the following config:
```
mp.messaging.outgoing."foo.name-with-dashes".connector=smallrye-kafka
mp.messaging.outgoing."foo.name-with-dashes".value.serializer=io.quarkus.kafka.client.serialization.ObjectMapperSerializer
mp.messaging.outgoing."foo.name-with-dashes".value.deserializer=io.quarkus.kafka.client.serialization.ObjectMapperDeserializer
```
The problem is that at some point I get double quotes inside this name. If I use a test resource like this:
```
public class Foo implements QuarkusTestResourceLifecycleManager {

  @Override
  public Map<String, String> start() {
    return Map.of(
        "mp.messaging.outgoing.\"\"treats.broadridge-raw-reports\"\".connector", "smallrye-kafka",
        "mp.messaging.outgoing.\"\"treats.broadridge-raw-reports\"\".value.serializer", "io.quarkus.kafka.client.serialization.ObjectMapperSerializer",
        "mp.messaging.outgoing.\"\"treats.broadridge-raw-reports\"\".value.deserializer", "io.quarkus.kafka.client.serialization.ObjectMapperDeserializer"
    );

  }

  @Override
  public void stop() {

  }
}
```
Everything works just fine. With this config I have the property keys twice in my application. At some point I get double quotes into the name.
I cant change the name of the topic.

### Expected behavior

My tests are green and kafka can start inside the test environment.

### Actual behavior

Kafka cant start because of property mismatch.

### How to Reproduce?

Use quarkus kafka properties with double quotes.

### Output of `uname -a` or `ver`

Linux ws7341vm.baadergroup.net 5.15.0-113-generic #123-Ubuntu SMP Mon Jun 10 08:16:17 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux

### Output of `java -version`

OpenJDK 64-Bit Server VM Temurin-21.0.5+11 (build 21.0.5+11-LTS, mixed mode, sharing)

### Quarkus version or git rev

3.28.5

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Gradle 9.0.0

### Additional information

_No response_

--- a/extensions/smallrye-reactive-messaging-kafka/deployment/src/main/java/io/quarkus/smallrye/reactivemessaging/kafka/deployment/SmallRyeReactiveMessagingKafkaProcessor.java
+++ b/extensions/smallrye-reactive-messaging-kafka/deployment/src/main/java/io/quarkus/smallrye/reactivemessaging/kafka/deployment/SmallRyeReactiveMessagingKafkaProcessor.java
@@ -124,8 +124,11 @@ public class SmallRyeReactiveMessagingKafkaProcessor {
     static String channelPropertyFormat = "mp.messaging.%s.%s.%s";
 
     static String getChannelPropertyKey(String channelName, String propertyName, boolean incoming) {
-        return String.format(channelPropertyFormat, incoming ? "incoming" : "outgoing",
-                channelName.contains(".") ? "\"" + channelName + "\"" : channelName, propertyName);
+        if ((channelName.charAt(0) != '"' || channelName.charAt(channelName.length() - 1) != '"')
+                && channelName.contains(".")) {
+            channelName = "\"" + channelName + "\"";
+        }
+        return String.format(channelPropertyFormat, incoming ? "incoming" : "outgoing", channelName, propertyName);
     }
 
     @BuildStep

